pkg/metrics/providers/graphite.go

```diff
@@ -60,6 +64,14 @@ func NewGraphiteProvider(provider flaggerv1.MetricTemplateProvider) (*GraphitePr
 		timeout: 5 * time.Second,
 	}
 
+	if username, ok := credentials["username"]; ok {
```

Comment:
The `PrometheusProvider` checks that `provider.SecretRef != nil` before attempting to read the username/password from `credentials`:
https://github.com/HBOCodeLabs/flagger/blob/main/pkg/metrics/providers/prometheus.go#L69-L81

While the `PrometheusProvider` is 1) similar to the `GraphiteProvider` in that username/password are _optional_ 2) perhaps somewhat of a canonical example, by nature of presumably being the first and likely most-widely-used provider, other providers seem to read `credentials`'s fields without first checking checking `provider.SecretRef`:

https://github.com/HBOCodeLabs/flagger/blob/main/pkg/metrics/providers/datadog.go#L81-L91

Is it better to check `provider.SecretRef`, or is the current `GraphiteProvider` implementation adequate?

Reply:
I think we would need to check for `provider.SecretRef` if basic auth is optional.

Reply:
@nmische Sounds good; I've added a `if provider.SecretRef == nil` check.